ql/src/test/queries/clientpositive/pointlookup6.q

```diff
@@ -0,0 +1,19 @@
+set hive.optimize.point.lookup.min=2;
+
+create table r_table (
+  string_col varchar(30)
+);
+
+create table l_table (
+  string_col varchar(14)
+);
+
+insert into r_table VALUES ('AAA111');
+insert into l_table VALUES ('AAA111');
+
+explain cbo
+SELECT l_table.string_col from l_table, r_table
+WHERE r_table.string_col = l_table.string_col AND l_table.string_col IN ('AAA111', 'BBB222') AND r_table.string_col IN ('AAA111', 'BBB222');
```

Comment:
If instead of `IN` we have an equivalent query with `OR` do we still hit the problem? Do we get the expected plan when the condition is expressed with `OR`?

Reply:
We get the expected plan in case of this query but I extended this patch by handling `OR` and added UT.